launch_ros/launch_ros/utilities/to_parameters_list.py

```diff
@@ -28,15 +28,31 @@
 from ..parameters_type import EvaluatedParameters
 
 
+def search_for_key(keys, dictionary):
+    """Search for the nested "ros__parameters" key and return the keys above it."""
+    for key, value in dictionary.items():
+        if key == "ros__parameters":
+            return (keys, value)
+        elif type(value) is dict:
+            keys.append(key)
+            return search_for_key(keys, value)
+        else:
+            return (None, None)
```

Comment:
@rebecca-butler hmm, what if there's several parameter groups in the same file? E.g.:

```
foo:
   node:
       ros__parameters:
             # ...
/**:
   ros__parameters:
       # ...
```

There's a precedence that has to be enforced here too. You may check [precedence in `rclcpp`](https://github.com/ros2/rclcpp/blob/d0cd6bb0a46f5b2841e9f282c9eecfa8cffe1ffa/rclcpp/src/rclcpp/detail/resolve_parameter_overrides.cpp#L56).